src/main/java/com/uber/cadence/worker/ShadowingWorker.java

```diff
@@ -0,0 +1,180 @@
+import com.uber.cadence.internal.shadowing.QueryBuilder;
+import com.uber.cadence.internal.shadowing.ReplayWorkflowActivity;
+import com.uber.cadence.internal.shadowing.ReplayWorkflowActivityImpl;
+import com.uber.cadence.internal.shadowing.ScanWorkflowActivity;
+import com.uber.cadence.internal.shadowing.ScanWorkflowActivityImpl;
+import com.uber.cadence.internal.sync.SyncActivityWorker;
+import com.uber.cadence.internal.worker.SingleWorkerOptions;
+import com.uber.cadence.internal.worker.Suspendable;
+import com.uber.cadence.serviceclient.IWorkflowService;
+import com.uber.cadence.shadower.WorkflowParams;
+import com.uber.cadence.shadower.shadowerConstants;
+import com.uber.cadence.workflow.Functions;
+import com.uber.m3.tally.Scope;
+import com.uber.m3.util.ImmutableMap;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public final class ShadowingWorker implements Suspendable {
+
+  private final IWorkflowService service;
+  private final SyncActivityWorker activityWorker;
+  private final ReplayWorkflowActivity replayActivity;
+  private final String taskList;
+  private final ShadowingOptions shadowingOptions;
+  private final AtomicBoolean started = new AtomicBoolean();
+
+  ShadowingWorker(
+      WorkflowClient client,
+      String taskList,
+      WorkerOptions options,
+      ShadowingOptions shadowingOptions) {
+    options = MoreObjects.firstNonNull(options, WorkerOptions.defaultInstance());
+    this.shadowingOptions = Objects.requireNonNull(shadowingOptions);
+    this.taskList = shadowingOptions.getDomain() + "-" + taskList;
+    this.service = client.getService();
+    Scope metricsScope =
+        client
+            .getOptions()
+            .getMetricsScope()
+            .tagged(ImmutableMap.of(MetricsTag.TASK_LIST, taskList));
+    ScanWorkflowActivity scanActivity = new ScanWorkflowActivityImpl(client.getService());
+    replayActivity = new ReplayWorkflowActivityImpl(client.getService(), metricsScope, taskList);
+
+    SingleWorkerOptions activityOptions =
+        SingleWorkerOptions.newBuilder()
+            .setIdentity(client.getOptions().getIdentity())
+            .setDataConverter(client.getOptions().getDataConverter())
+            .setTaskExecutorThreadPoolSize(options.getMaxConcurrentActivityExecutionSize())
+            .setTaskListActivitiesPerSecond(options.getTaskListActivitiesPerSecond())
+            .setPollerOptions(options.getActivityPollerOptions())
+            .setMetricsScope(metricsScope)
+            .build();
+    activityWorker =
+        new SyncActivityWorker(
+            client.getService(), client.getOptions().getDomain(), this.taskList, activityOptions);
+    activityWorker.setActivitiesImplementation(scanActivity, replayActivity);
+  }
+
+  void start() throws Exception {
+    if (!started.compareAndSet(false, true)) {
+      return;
+    }
+    startShadowingWorkflow();
```

Comment:
shall we validate shadow options here? e.g. make sure exit condition is set when using continuous mode.

Reply:
Added into options build()